homeassistant/components/bthome/__init__.py

```diff
@@ -43,6 +44,11 @@ def process_service_info(
         entry.entry_id
     ]
     discovered_device_classes = coordinator.discovered_device_classes
+    if entry.data.get(CONF_SLEEPY_DEVICE, False) != data.sleepy_device:
+        hass.config_entries.async_update_entry(
+            entry,
+            data=entry.data | {CONF_SLEEPY_DEVICE: data.sleepy_device},
+        )
```

Comment:
This is a bug in bthome since if the device is asleep we don't remember its sleepy.

xiaomi_ble has the same problem